pages/api/checkout/[sessionId].ts

```diff
@@ -0,0 +1,38 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import Stripe from "stripe";
+
+// set up stripe instance
+// TODO: set this as exportable variable in utils directory
+const stripe = new Stripe(process.env.STRIPE_SECRET_KEY, {
+    apiVersion: "2020-08-27",
+});
+
+/**
+ * sessionIdHandler takes the sessionId parameter and returns
+ * the Stripe session information associated with the sessionId
+ * @param req API request object
+ * @param res API response object
+ */
+export default async function sessionIdHandler(
+    req: NextApiRequest,
+    res: NextApiResponse,
+): Promise<void> {
+    if (req.method == "GET") {
+        // obtain the sessionId
+        const { sessionId } = req.query;
+
+        // obtain the session information from sessionId
+        const session = await stripe.checkout.sessions.retrieve(
+            sessionId as string,
+            {
+                expand: ["payment_intent"],
+            },
+        );
+
+        // return session information
+        res.status(200).json({ session });
+    } else {
+        // TODO: add JSON response for method not allowed
+        res.status(405);
```

Comment:
Maybe add the JSON response with ResponseObject as well similar to https://github.com/uwblueprint/social-diversity-for-children/pull/12/files

Also, consider setting the Allow header to a list of methods that are allowed? https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Allow

Reply:
Yeah - will do. I was waiting for your PR to be merged.

Reply:
added header - will address your first comment in a separate PR